pkg/server/service/service.go

```diff
@@ -161,9 +161,24 @@ func (m *Manager) getWRRServiceHandler(ctx context.Context, serviceName string,
 		if err != nil {
 			return nil, err
 		}
-
 		balancer.AddService(service.Name, serviceHandler, service.Weight)
+		if config.HealthCheck == nil {
+			continue
+		}
+		childName := service.Name
+		updater, ok := serviceHandler.(healthcheck.StatusUpdater)
+		if !ok {
+			log.FromContext(ctx).Debugf("Child service %v will not propagate to its parent %v any status change", service.Name, serviceName)
+			continue
+		}
+		log.FromContext(ctx).Debugf("Child service %v will update parent %v on status change", service.Name, serviceName)
+		// TODO(mpl): do we make RegisterStatusUpdater return an error (e.g. in case the
+		// updater actually does not want healthcheck enabled, according to its own config)
```

Comment:
As of now (but still TBD), we probably want to go that way, and straight up return here the value returned by RegisterStatusUpdater. This should make the service creation fail, which should be visible e.g. in the dashboard.